Provide default value for mark_cache_size

rschu1ze opened the pull request:
I played around with my local config.xml file. The minimal working example is this:

```
<?xml version="1.0"?>
<clickhouse>
    <mark_cache_size>5368709120</mark_cache_size>
    <listen_host>localhost</listen_host>
    <tcp_port>9000</tcp_port>
    <users_config>users.xml</users_config>
    <logger><console>true</console></logger>
</clickhouse>
```

Not specifying mark_cache_size made the server not start up:
```
  2022.05.18 12:15:06.549078 [ 8728320 ] {} <Error> Application: Not found: mark_cache_size
```

Looking at ClickHouse's ca. 100 server configuration options +
sub-options, it seems that mark_cache_size is NOT special enough to
require explicit configuration but instead that the behavior was
unintended because no default value was provided.

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Do no longer abort server startup if configuration option "mark_cache_size" is not explicitly set

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

rschu1ze committed:
Provide default value for mark_cache_size

I played around with my local config.xml file. The minimal working example is this:
<?xml version="1.0"?>
<clickhouse>
    <mark_cache_size>5368709120</mark_cache_size>
    <listen_host>localhost</listen_host>
    <tcp_port>9000</tcp_port>
    <users_config>users.xml</users_config>
    <logger><console>true</console></logger>
</clickhouse>

Not specifying mark_cache_size made the server not start up:
  2022.05.18 12:15:06.549078 [ 8728320 ] {} <Error> Application: Not found: mark_cache_size

Looking at ClickHouse's ca. 100 server configuration options +
sub-options, it seems that mark_cache_size is NOT special enough to
require explicit configuration but instead that the behavior was
unintended because no default value was provided.

azat commented on programs/server/Server.cpp:
```diff
@@ -1344,8 +1344,8 @@ int Server::main(const std::vector<std::string> & /*args*/)
-    /// Size of cache for marks (index of MergeTree family of tables). It is mandatory.
-    size_t mark_cache_size = config().getUInt64("mark_cache_size");
+    /// Size of cache for marks (index of MergeTree family of tables).
+    size_t mark_cache_size = config().getUInt64("mark_cache_size", 5368709120);
```
Worth removing it from embedded config? https://github.com/ClickHouse/ClickHouse/blob/ae0d64d8e155ed8ce79f0845cbdc2de7b19de8da/programs/server/embedded.xml#L16

rschu1ze replied:
Hm, good point. Guess the answer is yes :)

There is also a ton of config.xml templates used by tests ... will check if it's worth to change them too.

rschu1ze committed:
Remove mark_cache_size from further files

- changed config.xml/yaml files used by CH's own internal tests which
  are (hopefully) not sensitive to mark_cache_size being set or not

- further occurrences exist but changing them seems a bad idea (e.g.
  because they are in customer-provided data)
